### PR TITLE
[WIP] Babel & Webpack Upgrades

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

This upgrades Babel to version 7.x and webpack/webpacker to version 4.x. This work is good for the project in general, but it also lays the groundwork to fix the Storybook issue mentioned in #338.

## Related Tickets & Documents

- #338 
- #3490 
- #2034

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Yes!

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,21 +1,31 @@
 {
   "presets": [
-    ["env", { "modules": false, "targets": { "browsers": "> 1%", "uglify": true }, "useBuiltIns": true }],
-    "preact",
```
Comment (nickytonline) [inline .babelrc]: The preact built-in is no longer necessary.

File: Gemfile [hunk 2]
```diff
@@ -93,7 +93,7 @@ gem "twilio-ruby", "~> 5.25" # The official library for communicating with the T
 gem "twitter", "~> 6.2" # A Ruby interface to the Twitter API
 gem "uglifier", "~> 4.1" # Uglifier minifies JavaScript files
 gem "validate_url", "~> 1.0" # Library for validating urls in Rails
-gem "webpacker", "~> 3.5" # Use webpack to manage app-like JavaScript modules in Rails
+gem "webpacker", "~> 4.x" # Use webpack to manage app-like JavaScript modules in Rails
```
Comment (nickytonline) [inline Gemfile]: webpacker 4 gives us webpack 4.

File: config/webpack/environment.js [hunk 3]
```diff
@@ -8,27 +8,7 @@
 
 // module.exports = environment;
 
+const { environment } = require('@rails/webpacker');
+const webpack = require('webpack');
 
-const { environment } = require('@rails/webpacker')
-const webpack = require('webpack')
-
-environment.plugins.append(
```
Comment (nickytonline) [inline config/webpack/environment.js]: These are deprecated in v4 in favour of the optimizations section. By default, webpack 4 generates a vendors bundle now, so no need for a configuration for this.

**Comment by nickytonline** [comment]
The frontend builds properly, but webpacker complains about yarn.lock integrity issues even though they pass when I run it from the command line. Still investigating.

**Comment by nickytonline** [comment]
Things work, but there are some oddities. The editor loads twice for example. I have a feeling it's related to the code splitting, but need to dig a bit deeper still.

I'm a little busy at the moment, so if someone wants to team up on this, awesome. If not, this will just take a little while.

**Comment by nickytonline** [comment]
Just giving you a mention here @maestromac as I know you have already looked into this before. I can probably look at this again mid-October. Just a little swamped at the moment ;)

**Comment by maestromac** [comment]
No worries @nickytonline . I can actually team up with you on this. Mind if I contribute to this PR?

**Comment by nickytonline** [comment]
> No worries @nickytonline . I can actually team up with you on this. Mind if I contribute to this PR?

For sure. 👏🏻

**Comment by nickytonline** [comment]
No idea why this PR no longer points to my branch. Instead it points to "unknown repository. Closing in favour of #4598

**Pull request closed without merging**